model/lesson.js

```diff
@@ -122,6 +139,40 @@ async function getLessonById(data) {
     );
 }
 
+/*Fetches verses in a lesson based*/
+//ASSUMPTIONS: Verses in a Lesson are contigous
+async function getLessonVerses(data) {
+    console.log(data);
+    var invalid = utils.simpleValidation(data, {
+        lesson_id: "integer",
+    });
+    if (invalid) {
+        return invalid;
+    }
+
+    let lesson = await getLessonById(data);
+    if (!lesson.success) {
+        return lesson;
+    }
+
+    let lessonInfo = [];
+    let currentVerse = lesson.data[0].lesson_start_verse;
+    let numVerses = lesson.data[0].lesson_end_verse - lesson.data[0].lesson_start_verse;
+    for (let i = 0; i <= numVerses; i++) {
+        let temp = await verseInfo.getVerseInfo({ verse_id: currentVerse })
+        lessonInfo[i] = temp.data;
+        currentVerse++;
+    }
+
+    let res = utils.setResult(
+        { lessonInfo },
```

Comment:
Let's make sure this object also contains the metadata about the lesson (i.e. start and end verse, source, etc.) And check that these things are accurate in the tests please.

Reply:
sure will include lesson check